src/codewhisperer/util/codeWhispererSession.ts

```diff
@@ -70,6 +76,19 @@ class CodeWhispererSession {
     getCompletionType(index: number): CodewhispererCompletionType {
         return this.completionTypes.get(index) || 'Line'
     }
+
+    reset() {
+        this.sessionId = ''
+        this.requestIdList = []
+        this.startPos = new Position(0, 0)
+        this.startCursorOffset = 0
+        this.leftContextOfCurrentLine = ''
+        this.language = 'python'
+        this.triggerType = 'OnDemand'
+        this.recommendations = []
+        this.suggestionStates.clear()
+        this.completionTypes.clear()
```

Comment:
yea this is easier to read and understand instead of setting different  values across different places. 
for completeness, let's reset `requestContext` as well even though it shouldn't make difference.

Reply:
Will add,
currently this is only used in test, later on we can call this one when the last session finishes, if session is still a global state like this.